accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,305 @@
+Confidential Compute is the way to solve this problem.
+But that a VM run in confidential mode needs to be attested and checked on regular base.
+
+SUSE Manager should provide an attestation report for a system.
+Also a history of the past checks should be kept and shown.
+
+The attestation is a process of multiple checks which can also depend on the hardware and used environment.
+SUSE Manager should provide a framework for executing such checks, store the results in the database and show the latest and historic results.
+The number of checks should be easy extendable in future versions.
+
+
+# Detailed design
+[design]: #detailed-design
+
+Confidential Compute Attestation cannot be performed on all systems.
+It depends on the used hardware and environment where the instance is running on.
+In most cases it is not possible to detect if an attestation can be performed or not.
+
+This requires a per system configuration which should contain at least two configuration options:
+
+- Enable/Disable Confidential Compute Attestation (Checkbox)
+- Environment selection
+  - KVM Guest Attestation on AMD EPYC Milan CPUs
+  - KVM Guest Attestation on AMD EPYC Genoa CPUs
+  - Azure Attestation
+  - etc.
+
+The information about the environment result in pillar data which help to select the matching states which should be executed on the minion for this task.
+
+The attestation is a 2 step process.
+First collect the data and second to verify them and write the result.
+
+### Step One
+The first step execute a state on the target system to collect the report data.
+This is a state.apply on the target system. The selected environment information determine the state which is executed.
+
+These collected data are stored into a report table with a status field.
+The status can be:
+
+- pending
+- failed
+- succeeded
+
+The initial status is `pending` which indicate that the verification has not yet happen.
+
+Writing the state result into the report table could be generalized by taking the state IDs as keys and the returned result as values.
+Putting this into a map and store it as Json into a data column.
+
+We also have a mapping between environment type and expected verification/results.
+At the end of step one all the expected result checks are created in the result table with the status `pending`.
+
+Environment Types and Result Types should be ENUM types in Java.
+Also the expected mappings which environment type require which result type will life in Java.
+
+
+### Step two
+The second step is a process which run in a dedicated environment like a container which verify the result.
+
+Such an attestation container is responsible for one or more result types.
+The attestation container search regular for pending attestations of its result type.
+It read the data from the report table, performs all the needed checks and update the result table with the outcome
+and set the status to `succeeded` or `failed`.
```

Comment:
If some checks are failed and other are passed, would be great if we could also show this information to the user somewhere.

Reply:
The plan is to have a details page with a checklist.
This list show for every check if it was `succeeded` or `failed`.
The text behind the checkbox is the value `summary` from the result table. (maybe renamed to `title` or `description`